certbot-nginx/local-oldest-requirements.txt

```diff
@@ -1,3 +1,3 @@
 # Remember to update setup.py to match the package versions below.
 acme[dev]==0.29.0
-certbot[dev]==0.36.0
+-e certbot[dev]
```

Comment:
I think changing this to `-e .[dev]` should fix the oldest test failures. (Also, we shouldn't have to use `.` anymore in the future after step 2.a under "How do we express this?" in the doc is done.)

Also, don't forget to update `certbot-nginx/setup.py` to match the changes here.

Reply:
I actually think this should ideally be resolved by fixing the lock thing as well, if it's not I'll be surprised.

Reply:
ah jk this was a different error in the lock test

Reply:
why is there an external lock test being run under `tests/` by `nginx-oldest` anyway

Reply:
The test is a simple system test that we're locking files the way we're supposed to and like our other tests, it's run on both the "newest" and "oldest" setups to make sure everything still works. It's relevant for nginx because it also test the lockfile created by the nginx plugin.

Reply:
but the nginx plugin doesn't import lock.py...

Reply:
But it calls `certbot.util.lock_dir_until_exit` [here](https://github.com/certbot/certbot/blob/4b488614cf7749c8139c11f0983fe4b71e29827f/certbot-nginx/certbot_nginx/configurator.py#L210) which does internally.

If I remember correctly, I think you (rightly) pushed for `tests/lock_test.py` to test that all of our locks were always acquired in the same order.

Reply:
yep, still think that's correct. still weird that changes to internal naming would make the nginx tests break.

Reply:
I think the problem is that `tests/lock_test.py` imports `certbot.lock`. This is awkward, it probably shouldn't, and certbot-nginx doesn't actually require a newer version of Certbot, but I personally think just bumping the minimum version to fix this is fine.

Reply:
agreed on all counts.